dev_docs/tutorials/saved_objects.mdx

```diff
@@ -154,146 +155,104 @@ router.get(
- 1.4.0, we want to add a new id field to every panel with a newly generated UUID.
+When registering a SO type, a [modelVersions](https://github.com/elastic/kibana/blob/9a6a2ccdff619f827b31c40dd9ed30cb27203da7/packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts#L138-L177)
+property is available. This attribute is a map of version numbers to [SavedObjectsModelVersion](https://github.com/elastic/kibana/blob/9a6a2ccdff619f827b31c40dd9ed30cb27203da7/packages/core/saved-objects/core-saved-objects-server/src/model_version/model_version.ts#L12-L20)
+which is the top-level type/container to define model versions.
 
-First, the current mappings should always reflect the latest or "target" schema. Next, we should define a migration function for each step in the schema evolution:
+The modelVersion map is of the form `{ [version: number] => versionDefinition }`, using single integer to identify a version definition.
 
-**src/plugins/my_plugin/server/saved_objects/dashboard_visualization.ts**
+The first version must be numbered as version 1, incrementing by one for each new version.
 
 ```ts
-import { SavedObjectsType, SavedObjectMigrationFn } from 'src/core/server';
-import uuid from 'uuid';
+import { schema } from '@kbn/config-schema';
+import { SavedObjectsType } from 'src/core/server';
 
-interface DashboardVisualizationPre110 {
-  title: string;
-  subtitle: string;
-  panels: Array<{}>;
-}
-interface DashboardVisualization110 {
-  title: string;
-  panels: Array<{}>;
-}
 
-interface DashboardVisualization140 {
-  title: string;
-  panels: Array<{ id: string }>;
-}
-
-const migrateDashboardVisualization110: SavedObjectMigrationFn<
-  DashboardVisualizationPre110, [1]
-  DashboardVisualization110
-> = (doc) => {
-  const { subtitle, ...attributesWithoutSubtitle } = doc.attributes;
-  return {
-    ...doc, [2]
-    attributes: {
-      ...attributesWithoutSubtitle,
-      title: `${doc.attributes.title} - ${doc.attributes.subtitle}`,
-    },
-  };
-};
-
-const migrateDashboardVisualization140: SavedObjectMigrationFn<
-  DashboardVisualization110,
-  DashboardVisualization140
-> = (doc) => {
-  const outPanels = doc.attributes.panels?.map((panel) => {
-    return { ...panel, id: uuid.v4() };
-  });
-  return {
-    ...doc,
-    attributes: {
-      ...doc.attributes,
-      panels: outPanels,
-    },
-  };
-};
+const schemaV1 = schema.object({ title: schema.string({ maxLength: 50, minLength: 1 }) });
+const schemaV2 = schemaV1.extends({
+  description: schema.maybe(schema.string({ maxLength: 200, minLength: 1 })),
+});
 
 export const dashboardVisualization: SavedObjectsType = {
-  name: 'dashboard_visualization', [1]
-  /** ... */
-  migrations: {
-    // Takes a pre 1.1.0 doc, and converts it to 1.1.0
-    '1.1.0': migrateDashboardVisualization110,
-
-    // Takes a 1.1.0 doc, and converts it to 1.4.0
-    '1.4.0': migrateDashboardVisualization140, [3]
+  name: 'dashboard_visualization',
+  ...
+  mappings: {
```

Comment:
Relevant changes introduced by `f778b86a765c590eb9b0118595c8dcf3e2610623` -- apologies for whitespace noise 🤦🏻